src/core/socket.c

```diff
@@ -2356,6 +2356,8 @@ static void socket_enter_running(Socket *s, int cfd_in) {
                         }
 
                         r = manager_add_job(UNIT(s)->manager, JOB_START, UNIT_DEREF(s->service), JOB_REPLACE, &error, /* ret = */ NULL);
+                        if (r == -EDEADLK)
+                                return (void) log_unit_debug_errno(UNIT(s), r, "Failed to queue service startup job, ignoring: %s", bus_error_message(&error, r));
```

Comment:
I'm pretty sure this is wrong. The enqueuing is willy-nilly skipped without handling IO event in any way, hence it's going to activate again and again.

I've prepared #37505 as an alternative.